code/datums/components/attachment_handler.dm

```diff
@@ -100,6 +101,13 @@
 
 	RegisterSignal(attachment, COMSIG_ATOM_UPDATE_ICON, .proc/update_parent_overlay)
 	update_parent_overlay()
+	if(!CHECK_BITFIELD(attachment_data[FLAGS_ATTACH_FEATURES], ATTACH_APPLY_ON_MOB))
+		return
+	var/obj/parent_obj = parent
+	if(!ismob(parent_obj.loc))
+		return
+	var/mob/wearing_mob = parent_obj.loc
+	wearing_mob.regenerate_icons() ///Theres probably a better way to do this.
```

Comment:
what *are* you doing here

Reply:
When the item somethings attached to is worn, if i dont do regenerate_icons() you have to take the uniform off to get the new attachments to show. This acts like update_icon() as far as i can tell but i dont like it because it updates all of the mobs worn stuff.